soc/arm/nordic_nrf/nrf53/soc.c

```diff
@@ -224,6 +225,15 @@ static bool cpu_idle_prepare_monitor_end(void)
 	return __STREXB(0U, &cpu_idle_prepare_monitor_dummy);
 }
 
+static void rtc_pretick_finish_previous(void)
+{
+	NRF_IPC->PUBLISH_RECEIVE[CONFIG_SOC_NRF53_RTC_PRETICK_IPC_CH_TO_NET] &=
+			~IPC_PUBLISH_RECEIVE_EN_Msk;
+
+	nrf_rtc_event_clear(NRF_RTC1, NRF_RTC_CHANNEL_EVENT_ADDR(RTC1_PRETICK_CC_CHAN));
```

Comment:
What do you think about adding 
```c
NRF_WDT->TASKS_STOP = 1;
```
in this function as well?

Reply:
Stopping WDT happens from idle task when we are ok to sleep and the pretick CC did not fire yet.

Reply:
Stopping it here could lower the power consumption and wouldn't change the solution functionally. Let's proceed with what we have though to minimize the risk that some additional race or bug slips through when we stop the WDT here.